### PR TITLE
ci: Constrain release trigger to be semantic

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes #N/A 

**Description**

Constrain release tagging to be semantic

**How was this change tested?**

- Ran Github Actions on local repo to test pattern matching

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -1,7 +1,12 @@
 name: Release
 on:
   push:
-    tags: [ 'v*.*.*' ]
+    # Only release on supported semantic version tagging e.g. v0.0.1-rc.0
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
```
Comment (tzneal) [inline .github/workflows/release.yaml:6]: Should the . be escaped?

Comment (jonathan-innis) [reply .github/workflows/release.yaml:6]: Glob patterns aren't regex. They don't get escaped: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

Comment (tzneal) [reply .github/workflows/release.yaml:6]: TIL

**Review by tzneal — APPROVED** [review]
Lgtm

**Pull request merged**